src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -829,12 +829,22 @@ private void LaunchStreams(ConnectionState connectionState, Dictionary<string, o
             {
                 _ = _sendIAsyncStreamItemsMethod
                     .MakeGenericMethod(reader.GetType().GetInterface("IAsyncEnumerable`1")!.GetGenericArguments())
-                    .Invoke(this, new object[] { connectionState, kvp.Key.ToString(), reader, cts });
+                    .Invoke(this, [connectionState, kvp.Key.ToString(), reader, cts]);
                 continue;
             }
-            _ = _sendStreamItemsMethod
-                .MakeGenericMethod(reader.GetType().GetGenericArguments())
-                .Invoke(this, new object[] { connectionState, kvp.Key.ToString(), reader, cts });
+            else
+            {
+                if (ReflectionHelper.TryGetStreamType(reader.GetType(), out var channelGenericType))
+                {
+                    _ = _sendStreamItemsMethod
+                        .MakeGenericMethod(channelGenericType)
+                        .Invoke(this, [connectionState, kvp.Key.ToString(), reader, cts]);
```

Comment:
If this is the only place we use the `_streamTypeLookup` cache, we might as well cache the closed `MethodInfo` returned by `MakGenericMethod` or better yet the `Func<ConnectionState, string, ChannelReader<T>, CancellationTokenSource, Task>` we build from a compiled `Expression`. Since we're just trying to get the dependency update merged, we can just forget the cache for now.